cpp/src/parquet/column_reader.h

```diff
@@ -368,6 +368,11 @@ class PARQUET_EXPORT RecordReader {
 
   virtual void DebugPrintState() = 0;
 
+  /// \brief Returns the dictionary owned by the current decoder. Throws an
+  /// exception if the current decoder is not for dictionary encoding.
+  /// \param[out] dictionary_length The number of dictionary entries.
+  virtual const uint8_t* ReadDictionary(int32_t* dictionary_length) = 0;
```

Comment:
why uint8_t should this be int32?  or provide some more details on how to interpret this?

Reply:
It seems that comment of `values()` says `FLBA and ByteArray types do not use this array`. Should we change this into a template function or simply return `const void*` so downstream is required to be aware of the type to avoid misuse the `uint8_t*`?